src/tiled_dataset.py

```diff
@@ -25,59 +25,27 @@ def __init__(
-        self.mask_client = from_uri(mask_tiled_uri, api_key=mask_tiled_api_key)
-        self.mask_idx = list(mask_idx)
-        self.shift = int(shift)
-        self.transform = transform
-
-    def __len__(self):
-        return len(self.mask_client)
-
-    def __getitem__(self, idx):
-        data = self.data_client[self.mask_idx[idx],]
-        mask = self.mask_client[idx,].astype('int') - self.shift # Conversion to int is needed as switching unlabeled pixels to -1 would cause trouble in uint8 format
-        if self.transform:
-            return self.transform(data), mask
+        if mask_tiled_uri:
+            self.mask_client = from_uri(mask_tiled_uri, api_key=mask_tiled_api_key)
         else:
-            return data, mask
-        
-class InferenceDataset(torch.utils.data.Dataset):
-    def __init__(
-            self,
-            data_tiled_uri,
-            mask_tiled_uri,
-            seg_tiled_uri,
-            mask_idx,
-            data_tiled_api_key=None,
-            mask_tiled_api_key=None,
-            seg_tiled_api_key=None,
-            transform=None):
-        '''
-        Args:
-            data_tiled_uri:      str,    Tiled URI of the input data
-            mask_tiled_uri:      str,    Tiled URI of mask
-            seg_tiled_uri:       str,    Tiled URI of segmentation results
-            mask_idx:            list,   List for slice indexs that maps the mask to corresponding input images, used for slicing here  
-            data_tiled_api_key:  str,    Tiled API key for input data access
-            mask_tiled_api_key:  str,    Tiled API key for mask access
-            seg_tiled_api_key:   str,    Tiled API key for segmentation results
-            transform:           callable, if not given return PIL image
-
-        Return:
-            ml_data:             data_tensor
-        '''
-        self.data_client = from_uri(data_tiled_uri, api_key=data_tiled_api_key)
-        self.mask_client = from_uri(mask_tiled_uri, api_key=mask_tiled_api_key)
-        self.seg_client = from_uri(seg_tiled_uri, api_key=seg_tiled_api_key)
-        self.mask_idx = list(mask_idx)
+            self.mask_client = None
+        self.mask_idx = list(mask_idx) if mask_idx else []
+        self.shift = int(shift)
         self.transform = transform
 
     def __len__(self):
         return len(self.mask_idx)
 
     def __getitem__(self, idx):
         data = self.data_client[self.mask_idx[idx],]
-        if self.transform:
-            return self.transform(data)
+        if self.mask_client:
+            mask = self.mask_client[idx,].astype('int') - self.shift # Conversion to int is needed as switching unlabeled pixels to -1 would cause trouble in uint8 format
+            if self.transform:
+                return self.transform(data), mask
+            else:
+                return data, mask
         else:
-            return data
\ No newline at end of file
+            if self.transform:
+                return self.transform(data)
```

Comment:
Just to double check, this is also how DLSIA's internal inference methods that we may use later on expect data to be passed (i.e. not still a tuple)?

Reply:
As discussed in person we are good for this part as the inference function only asks for the image part, not the pair.